sphinx/ext/autosummary/generate.py

```diff
@@ -90,12 +91,19 @@ def _underline(title, line='='):
 
 # -- Generating output ---------------------------------------------------------
 
-def generate_autosummary_docs(sources, output_dir=None, suffix='.rst',
-                              warn=_simple_warn, info=_simple_info,
-                              base_path=None, builder=None, template_dir=None,
-                              imported_members=False, app=None):
-    # type: (List[str], str, str, Callable, Callable, str, Builder, str, bool, Any) -> None
-
+def generate_autosummary_docs(sources,                 # type: List[str]
+                              output_dir=None,         # type: str
+                              suffix='.rst',           # type: str
+                              warn=_simple_warn,       # type: Callable
+                              info=_simple_info,       # type: Callable
+                              base_path=None,          # type: str
+                              builder=None,            # type: Builder
+                              template_dir=None,       # type: str
+                              imported_members=False,  # type: bool
+                              recursion_limit=0,       # type: int
```

Comment:
Could you add this to the tail of argument list to keep compatibility? I'm afraid of changing the order of parameters.

Reply:
done